### PR TITLE
AMPI: Fail compilation when unable to get binary path

### PR DESCRIPTION
**Pull request by matthiasdiener**
In this case, any AMPI application would abort anyway on startup, so better to fail at AMPI compilation already.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Sure, but if I understand the current code correctly, `getAmpiBinaryPath` is called on startup of each AMPI application, independent of whether it uses pipglobals or fsglobals. Are you suggesting to change the `getAmpiBinaryPath` call to only execute when pipglobals or fsglobals are requested?

**Comment by evan-charmworks** [comment]
I think I made a mistake by putting an abort in this location. pipglobals and fsglobals still need to fail if no support is available here (and compile time would be better than runtime) but AMPI in general does not.

I'm not sure of the best way to only get the binary path when it is needed.

**Comment by matthiasdiener** [comment]
IMHO we could put `#error` instead of the `CkAbort()`, like in this PR. It is very unlikely that the `#else` case gets executed, and this would most likely indicate some configuration error, not lack of support in the system. At least, this is how I found the issue (missed configure update when working on #2142).

**Comment by matthiasdiener** [comment]
Got hit again by this bug when working on the cmake build system...

File: src/libs/ck-libs/ampi/ampi.C [hunk 1]
```diff
@@ -987,7 +987,7 @@ static void getAmpiBinaryPath() noexcept
     ampi_binary_path[n] = '\0';
   }
 #else
-  CkAbort("Could not get path to current binary!");
+#  error "AMPI: No known way to get path to current binary."
```
Comment (stwhite91) [inline src/libs/ck-libs/ampi/ampi.C]: Could add a comment somewhere here that says we only really need the binary path when using pipglobals or fsglobals.

Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampi.C]: done

**Pull request merged**